src/history/historyActions.ts

```diff
@@ -28,9 +30,9 @@ export const getHistoryData: (
   symbol: string
 ) => ThunkResult<void, HistoryReceivedAction> = symbol => (dispatch, getState, api) => {
   const { priceFilter, dateFilter } = getState().history
-  api.getHistory(symbol, dateFilter, priceFilter).then((history: HistoryData[]) => {
-    return dispatch(historyReceivedAction(history))
-  })
+  return api
+    .getHistory(symbol, dateFilter, priceFilter)
+    .then((history: HistoryData[]) => dispatch(historyReceivedAction(history)))
```

Comment:
you don't actually need the `: HistoryData[]` type annotation here as it will be inferred, assuming `getHistory` is typed correctly.